plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -558,6 +567,96 @@ public void updatePartitionStatistics(Table table, String partitionName, Functio
         removedStatistics.forEach(column -> deletePartitionColumnStatistics(table.getDbName(), table.getTableName(), partitionName, column));
     }
 
+    @Override
+    public void updatePartitionStatistics(Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
+    {
+        if (updatePartitionStatisticsBatchEnabled) {
+            if (updates.size() <= updatePartitionStatisticsBatchSize) {
+                updatePartitionStatisticsBatch(table, updates);
+                return;
+            }
+
+            Map<String, Function<PartitionStatistics, PartitionStatistics>> batch = new HashMap<>(updatePartitionStatisticsBatchSize);
+            updates.forEach((partitionName, update) -> {
+                batch.put(partitionName, update);
+                if (batch.size() >= updatePartitionStatisticsBatchSize) {
+                    updatePartitionStatisticsBatch(table, batch);
+                    batch.clear();
+                }
+            });
+            if (!batch.isEmpty()) {
+                updatePartitionStatisticsBatch(table, batch);
+                batch.clear();
+            }
+        }
+        else {
+            updates.forEach((partitionName, update) -> updatePartitionStatistics(table, partitionName, update));
+        }
+    }
+
+    private void updatePartitionStatisticsBatch(Table table, Map<String, Function<PartitionStatistics, PartitionStatistics>> updates)
+    {
+        List<String> partitionNames = updates.keySet().stream().collect(toImmutableList());
+        List<Partition> partitions = getPartitionsByNames(table.getDbName(), table.getTableName(), partitionNames);
+        checkState(partitionNames.size() == partitions.size(),
+                "getPartitionsByNames() did not return required partition count, required %s, actual %s",
+                partitionNames.size(), partitions.size());
+
+        Map<String, PartitionStatistics> partitionStatistics = getPartitionStatistics(table, partitions);
+
+        ImmutableList.Builder<Partition> modifiedPartitionBatchBuilder = ImmutableList.builder();
+        ImmutableMap.Builder<String, List<ColumnStatisticsObj>> updatedStatisticsBatchBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<String, Set<String>> removedStatisticsBatchBuilder = ImmutableMap.builder();
+
+        for (int i = 0; i < partitionNames.size(); i++) {
+            String partitionName = partitionNames.get(i);
+            Partition originalPartition = partitions.get(i);
+            Function<PartitionStatistics, PartitionStatistics> update = updates.get(partitionName);
+
+            PartitionStatistics currentStatistics = requireNonNull(
+                    partitionStatistics.get(partitionName), "getPartitionStatistics() did not return statistics for partition");
+            PartitionStatistics updatedStatistics = update.apply(currentStatistics);
+
+            Partition modifiedPartition = originalPartition.deepCopy();
```

Comment:
Consider reusing the code from `io.trino.plugin.hive.metastore.thrift.ThriftHiveMetastore#updatePartitionStatistics(org.apache.hadoop.hive.metastore.api.Table, java.lang.String, java.util.function.Function<io.trino.plugin.hive.PartitionStatistics,io.trino.plugin.hive.PartitionStatistics>)`